### ISSUE TITLE
blue_green_terminate_inactive

### ISSUE THREAD

**Issue by bryantidd** [ThoughtWorksStudios/eb_deployer#55]
When setting blue_green_terminate_inactive under an environment, it seems to try and remove the last inactive application version not the actual environment stack that is inactive. This is a major issue when you have multiple environments in the same AWS account. It also seems that nothing actually gets terminated.
I'm not a ruby developer or I would provide a fix, but I am a user of this gem and would appreciate this being looked into. Otherwise I will have to find another way of removing inactive stacks by environment.
Thanks you,
Bryan

**Comment by bryantidd** [comment]
Could be an old version issue.

**Comment by bryantidd** [comment]
Nope, This doesn't work! It removes an application version, but doesn't remove the inactive stack.
I'm not convinced it even removes the correct application version. This is a major issue! Please address!

**Comment by wpc** [comment]
hi, @bryantidd , thanks for reporting this. I will take a look tonight.

**Comment by wpc** [comment]
Hi, @bryantidd 

My bad. I forget to release a new version after merge the PR. Please upgrade to 0.6.2 and have a try.

**Comment by bryantidd** [comment]
Looks good, one issue with blue/green is that if you end up with an inactive stack in an error state, and try to rerun over it, it fails to remove the inactive stack because it may at the time of eval not been in a green state...to work around this issue, should a remove_stack_on_failed_deploy be added. They way automated systems that don't care why something failed, just that it did and we need to roll forward to a clean state for the next build...would have a flag to work off of. Any thoughts...

**Comment by wpc** [comment]
hi, @bryantidd:

Sorry for the delayed response, somehow I missed this your comment, bad on me. I have hard time to understand: "inactive stack in an error state, and try to rerun over it, it fails to remove the inactive stack because it may at the time of eval not been in a green state". Do you mean the eb environment removal in the Phoenix server mode failed?

**Comment by bryantidd** [comment]
No, sorry, it's basically that sometimes due to an error in the interaction between EC2 and ElasticBeanstalk security groups do not get removed, so the stack termination fails. It would be cool if we had a retry termination capability on the blue green switch to help us with this until Amazon resolves their problem...it's been ongoing issue with them for a long time.

**Comment by davidojedalopez** [comment]
I can not find any documentation about the 'blue_green_terminate_inactive' setting you are talking about @bryantidd. Where can I find it? Or how can I use it? It would really help me out since I do not want to keep an environment inactive wasting resources (I already configured to remove instances, but the load balancer remains in the inactive environment)
